lib/decoder.js

```diff
@@ -281,12 +281,24 @@ Decoder.prototype.decodeUint = function(offset, size) {
 
 Decoder.prototype.decodeBigUint = function(offset, size) {
   var buffer = new Buffer(size);
+  var hasBigInt = typeof BigInt === 'function';
```

Comment:
I think it should be done once, and configured as a helper fn at `Decoder` initialization

Something like

```
this.bigIntHelper = typeof BigInt === 'function' ?
   nativeBigIntHandler : 
   thirdPartyBigIntHandler
```

And then just use it where needed - it'll also help with testing.

Reply:
Good call.  I didn't end up writing a helper function because the arithmetic APIs differ, so for now it's just a boolean.